### PR TITLE
feat: dataset removal

### LINKED ISSUE #118: User can remove a dataset
Perhaps under the 'Dataset' OS menu, 'Remove'

From 9/6 demo:
"spruce up language that says this will remove all versions of the dataset on the qri database. check box: “Also remove the dataset files from my file system”"

### PR DESCRIPTION
**Pull request by chriswhong**
Screenshot_9_12_19__7_06_PM
- Adds a contextual menu on dataset list items for 'Remove...' (Closes #118)
- Adds a remove dataset modal, asking the user whether they would like remove files (if the dataset is fsi linked)
- Refactors modal types, and the implementation of modals in App.tsx.  All are consolidated under a single CSS transition, and are mounted based on the type of Modal that is currently set in state.

### CODE REVIEW

**Comment by b5** [comment]
just tried to remove an unlinked dataset. desktop wasn't happy

File: app/components/form/CheckboxInput.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import * as React from 'react'
+
+export interface CheckboxInputProps {
+  label?: string
+  name: string
+  checked: boolean
+  onChange: (name: string, checked: any) => void
+}
+
+const CheckboxInput: React.FunctionComponent<CheckboxInputProps> = ({ label, name, checked, onChange }) => {
```
Comment (b5) [inline app/components/form/CheckboxInput.tsx]: lovely

File: app/components/App.tsx [hunk 2]
```diff
@@ -48,6 +49,8 @@ export interface AppProps {
   setApiConnection: (status: number) => Action
   pingApi: () => Promise<ApiAction>
   setModal: (modal: Modal) => Action
+  removeDatasetAndFetch: (peername: string, name: string, removeFiles: boolean) => Promise<ApiAction>
```
[the remark this answers is not in the document]
Comment (chriswhong) [reply app/components/App.tsx]: +1

**Comment by b5** [comment]
Doooooope

**Pull request merged**